spec/features/characters/new_spec.rb

```diff
@@ -50,7 +69,62 @@
     end
   end
 
-  scenario "Creating character with icon, description and extant template"
+  scenario "Creating character with icon, description and extant template", js: true do
+    user = login
+    icon = create(:icon, user: user, keyword: 'Example icon')
+    template = create(:template, user: user, name: 'Example template')
+    visit new_character_path(template_id: template.id)
+
+    expect(page).to have_no_selector('.flash.error')
+    expect(page).to have_selector('#select2-character_template_id-container', text: 'Example template')
+
+    find("img[alt='Example icon']").click
+    within('.form-table') do
+      fill_in 'Character Name', with: 'Example character'
+      fill_in 'Template Nickname', with: 'Example nickname'
+      fill_in 'Screen Name', with: 'example_screenname'
+      fill_in 'Facecast', with: 'Example facecast'
+      fill_in 'Description', with: 'Example description'
+      click_button 'Save'
+    end
+    expect(page).to have_no_selector('.flash.error')
+    expect(page).to have_selector('.flash.success')
+    within('.flash.success') do
+      expect(page).to have_text('Character saved successfully.')
+    end
+
+    within('.character-info-box') do
+      expect(page).to have_selector('.character-name', text: 'Example character')
+      expect(page).to have_selector('.character-screenname', text: 'example_screenname')
+      expect(page).to have_selector('.character-icon')
```

Comment:
Can you check that this character-icon has the correct alt text? Just to be sure it used the clicked icon when producing the character.

Reply:
I think checking the id from the link should do that already, actually, but now it checks the alt text too I guess?? xD